src/main/kotlin/uk/gov/dluhc/printapi/database/entity/PrintDetails.kt

```diff
@@ -16,20 +16,23 @@ data class PrintDetails(
     var applicationReference: String? = null,
     @get:DynamoDbSecondaryPartitionKey(indexNames = [SOURCE_TYPE_GSS_CODE_INDEX_NAME]) var sourceType: SourceType? = null,
     var vacNumber: String? = null,
+    var vacVersion: String? = "1",
     var requestDateTime: OffsetDateTime? = null,
     var firstName: String? = null,
     var middleNames: String? = null,
     var surname: String? = null,
     var certificateLanguage: CertificateLanguage? = null,
+    var certificateFormat: CertificateFormat? = CertificateFormat.STANDARD,
     var photoLocation: String? = null,
     var delivery: CertificateDelivery? = null,
     @get:DynamoDbSecondarySortKey(indexNames = [SOURCE_TYPE_GSS_CODE_INDEX_NAME]) var gssCode: String? = null,
     var issuingAuthority: String? = null,
     var issueDate: LocalDate = LocalDate.now(),
+    var suggestedExpiryDate: LocalDate = issueDate.plusYears(10),
```

Comment:
I've not been close enough to the dynamo stuff (in both print-api and notifications-api), but I'm a bit confused over a couple of things ...

These dynamo entity (item ?) classes appear to be data classes, not regular classes that we used for JPA. Was that a conscious decision?

I'm also confused by all the nullable fields, even those with a default value. I assumed/guessed it worked the same way as JPA when retrieving records from the DB, in that JPA will new up the empty object with its no-args constructor, then will call the setters / use reflection to populate the fields.
So I assumed it worked the same way ....

But then I saw `issueDate`, `suggestedExpiryDate` and `status` are all non-nullable fields but with default values.
Was this intentional? Why are these fields different to the other fields, even those that also have a default value (eg: `vacVersion` and `certificateFormat`)

Reply:
I don't really think we can compare DynamoDB with JPA - there are no proxies, no dirty detection, no lazy loading, no recommendation that classes need to be mutable.  [Tutorials I read at the time](https://betterprogramming.pub/aws-java-sdk-v2-dynamodb-enhanced-client-with-kotlin-spring-boot-application-f880c74193a2) all used mutable data classes.

I added the defaults here as we don't currently have clear requirements for these properties and assume later stories will enhance/update/refactor the code appropriately - I saw that `IssueDate` had set it's default in similar manner so went with that approach.